code/modules/client/persistent_client.dm

```diff
@@ -1,55 +1,86 @@
 
-///assoc list of ckey -> /datum/player_details
-GLOBAL_LIST_EMPTY_TYPED(player_details, /datum/player_details)
+///assoc list of ckey -> /datum/persistent_client
+GLOBAL_LIST_EMPTY_TYPED(persistent_clients_by_ckey, /datum/persistent_client)
+/// A flat list of all persistent clients, for her looping pleasure.
+GLOBAL_LIST_EMPTY_TYPED(persistent_clients, /datum/persistent_client)
 
 /// Tracks information about a client between log in and log outs
-/datum/player_details
+/datum/persistent_client
+	/// The true client
+	var/client/client
+	/// The mob this pclient is currently bound to.
+	var/mob/mob
+
+	/// Major version of BYOND this client is using.
+	var/byond_version
+	/// Build number of BYOND this client is using.
+	var/byond_build
+
 	/// Action datums assigned to this player
 	var/list/datum/action/player_actions = list()
 	/// Tracks client action logging
 	var/list/logging = list()
+
 	/// Callbacks invoked when this client logs in again
 	var/list/post_login_callbacks = list()
 	/// Callbacks invoked when this client logs out
 	var/list/post_logout_callbacks = list()
+
 	/// List of names this key played under this round
 	/// assoc list of name -> mob tag
 	var/list/played_names = list()
 	/// Lazylist of preference slots this client has joined the round under
 	/// Numbers are stored as strings
 	var/list/joined_as_slots
-	/// Major version of BYOND this client is using.
-	var/byond_version
-	/// Build number of BYOND this client is using.
-	var/byond_build
+
 	/// Tracks achievements they have earned
 	var/datum/achievement_data/achievements
+
 	/// World.time this player last died
 	var/time_of_death = 0
 
-/datum/player_details/New(key)
-	achievements = new(key)
+/datum/persistent_client/New(ckey, client)
+	src.client = client
+	achievements = new(ckey)
+	GLOB.persistent_clients_by_ckey[ckey] = src
+	GLOB.persistent_clients += src
+
+/datum/persistent_client/Destroy(force)
+	SHOULD_CALL_PARENT(FALSE)
+	. = QDEL_HINT_LETMELIVE
+	CRASH("Who the FUCK tried to delete a persistent client? Get your head checked you leadskull.")
+
+/// Setter for the mob var, handles both references.
+/datum/persistent_client/proc/SetMob(mob/new_mob)
```

Comment:
why are all your new procs PascalCase?

Reply:
this one doesnt have to be i just felt like it